code/modules/clothing/glasses/hud.dm

```diff
@@ -45,17 +49,75 @@
 
 /obj/item/clothing/glasses/hud/emp_act(severity)
 	. = ..()
-	if(obj_flags & EMAGGED || . & EMP_PROTECT_SELF)
+	if(obj_flags & OBJ_EMPED || . & EMP_PROTECT_SELF)
 		return
-	obj_flags |= EMAGGED
+	obj_flags |= OBJ_EMPED
 	desc = "[desc] The display is flickering slightly."
+	addtimer(CALLBACK(src, .proc/reset_emp), rand(1200 / severity, 600 / severity))
+	//If we aren't glitching out already, start glitching
+	if(!(obj_flags & EMAGGED))
+		start_glitch()
+
+/obj/item/clothing/glasses/hud/proc/reset_emp()
+	obj_flags &= ~OBJ_EMPED
+	//If we aren't emagged, stop glitching
+	if(obj_flags & EMAGGED)
+		return
+	desc = initial(desc)
+	stop_glitch()
 
 /obj/item/clothing/glasses/hud/emag_act(mob/user)
 	if(obj_flags & EMAGGED)
 		return
 	obj_flags |= EMAGGED
 	to_chat(user, "<span class='warning'>PZZTTPFFFT</span>")
 	desc = "[desc] The display is flickering slightly."
+	//If we aren't already glitching out, start glitching
+	if(!(obj_flags & OBJ_EMPED))
+		start_glitch()
+
+/obj/item/clothing/glasses/hud/proc/start_glitch()
+	if(ismob(loc))
+		var/mob/M = loc
+		//Remove old glitching hud
+		if(glitching_hud)
+			glitching_hud.transform = matrix()
```

Comment:
Why do you need to transform it when you're going to null it later?

Reply:
Also can't you just qdel null it?

Reply:
could just null it and let GC deal with it, can't qdel matrices afaik

Reply:
Ok just took a look at the code and the reason we do this is:
The value of glitching_hud is still used by the mob. Deleting it would cause the entire hud to break.
If we just set the glitching_hud to null, then we remove the reference to it however the values we altered will stay the same, so we need to set them to null as well.

Reply:
Also using varedit in game shows the default value of transform to be /matrix, so I think it has to be a matrix instead of null.